Microsoft.Azure.Cosmos/src/HttpClient/CosmosHttpClientCore.cs

```diff
@@ -106,12 +107,18 @@ internal sealed class CosmosHttpClientCore : CosmosHttpClient
             // TODO: Remove type check and use #if NET6_0_OR_GREATER when multitargetting is possible
             Type socketHandlerType = Type.GetType("System.Net.Http.SocketsHttpHandler, System.Net.Http");
 
-            if (socketHandlerType != null)
+            try
             {
-                return CosmosHttpClientCore.CreateSocketsHttpHandlerHelper(gatewayModeMaxConnectionLimit, webProxy, serverCertificateCustomValidationCallback);
+                if (socketHandlerType != null)
+                {
+                    return CosmosHttpClientCore.CreateSocketsHttpHandlerHelper(gatewayModeMaxConnectionLimit, webProxy, serverCertificateCustomValidationCallback);
+                }
             }
-            
-            return CosmosHttpClientCore.CreateHttpClientHandlerHelper(gatewayModeMaxConnectionLimit, webProxy, serverCertificateCustomValidationCallback);
+            catch (Exception e) 
+            {
+                DefaultTrace.TraceError("Failed to create SocketsHttpHandler: {0}", e.ToString());
```

Comment:
```suggestion
                DefaultTrace.TraceError("Failed to create SocketsHttpHandler: {0}", e);
```

Don't materialize strings before hand